sqlglot/parser.py

```diff
@@ -7083,35 +7093,109 @@ def _parse_alter(self) -> exp.Alter | exp.Command:
+            # Empty kind  https://prestodb.io/docs/current/sql/analyze.html
+            this = self._parse_table_parts()
 
-        if self._match_text_seq("COMPUTE", "STATISTICS"):
-            return self.expression(
-                exp.Analyze,
-                kind=kind,
-                this=this,
-                partition=partition,
-                expression=self._parse_compute_statistics(),
+        partition = self._parse_partition()
+        inner_expression: t.Optional[exp.Expression] = None
+        if self._match_texts(self.ANALYZE_EXPRESSION_PARSERS):
+            inner_expression = self.ANALYZE_EXPRESSION_PARSERS[self._prev.text.upper()](self)
+
+        properties = self._parse_properties()
+        return self.expression(
+            exp.Analyze,
+            kind=kind,
+            this=this,
+            partition=partition,
+            properties=properties,
+            expression=inner_expression,
+            options=options,
+        )
+
+    # https://spark.apache.org/docs/3.5.1/sql-ref-syntax-aux-analyze-table.html
+    def _parse_analyze_statistics(self, kind) -> exp.Statistics:
+        this = None
+        option = self._prev.text.upper() if self._match_text_seq("DELTA") else None
+        expressions = []
+
+        if not self._match_text_seq("STATISTICS"):
+            self.raise_error("Expecting token STATISTICS")
+
+        if self._match_text_seq("NOSCAN"):
+            this = "NOSCAN"
+        elif self._match(TokenType.FOR):
+            if self._match_text_seq("ALL", "COLUMNS"):
+                this = "FOR ALL COLUMNS"
+            if self._match_texts("COLUMNS"):
+                this = "FOR COLUMNS"
+                expressions = self._parse_csv(self._parse_column_reference)
+        elif self._match_text_seq("SAMPLE"):
+            sample = self._parse_number()
+            expressions = (
+                [self.expression(exp.Command, this="SAMPLE", expression=f" {sample} PERCENT")]
+                if self._match(TokenType.PERCENT)
+                else []
             )
 
-        return self._parse_as_command(start)
+        return self.expression(
+            exp.Statistics, kind=kind, option=option, this=this, expressions=expressions
+        )
+
+    # https://dev.mysql.com/doc/refman/8.4/en/analyze-table.html
+    def _parse_analyze_histogram(self, this: str) -> exp.Histogram:
+        expression: t.Optional[exp.Expression] = None
+        expressions = []
+        update_options = None
+
+        if self._match_text_seq("HISTOGRAM", "ON"):
+            expressions = self._parse_csv(self._parse_column_reference)
+            if self._match(TokenType.WITH):
+                buckets = self._parse_number()
+                with_expression = (
+                    f"{buckets} {self._prev.text.upper()}"
+                    if self._match_text_seq("BUCKETS")
+                    else None
+                )
+                expression = self.expression(exp.With, expressions=[with_expression])
```

Comment:
Why is this instantiating a `With` expression? `With` is used to represent CTEs.